lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -31,29 +37,68 @@
 @DriverSide
 @Private
 @Unstable
-public interface DriverRestartManager {
+public final class DriverRestartManager {
```

Comment:
It might make sense to declare all methods in this class `synchronized` to avoid race conditions. A finer grained approach would be feasible, but I don't think this is in a performance-critical path.